src/Symfony/Component/Translation/CatalogueBag/FallbackCatalogueBag.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\CatalogueBag;
+
+use Symfony\Component\Translation\MessageCatalogueInterface;
+use Symfony\Component\Translation\MessageCatalogue;
+use Symfony\Component\Translation\Translator;
+
+/**
+ * Merges the fallback catalogues into the loaded one.
+ *
+ * @author Abdellatif Ait boudad <a.aitboudad@gmail.com>
+ */
+class FallbackCatalogueBag implements CatalogueBagInterface
```

Comment:
There is no "main" CatalogueBag. This is perfectly legit to move the fallback logic in a dedicated class.